src/engine.ml

```diff
@@ -938,10 +939,12 @@ let server_send_push_reply config is_not_taken tls session key tls_data =
     (* XXX(reynir): here we assume the binding for [Key_derivation] is only ever [`Tls_ekm] *)
     @ (if Config.mem Key_derivation config then [ "key-derivation tls-ekm" ]
        else [])
-    @ (Option.to_list (Config.find Protocol_flags config)
-      |> List.map (fun flags ->
-             Fmt.str "%a" (Config.pp_b ?sep:None)
-               (Config.Conf_map.B (Protocol_flags, flags))))
+    @ (Config.find Protocol_flags config
+      |> Option.map (fun flags ->
+             Fmt.str "%a" (Config.pp_b ~sep)
```

Comment:
Why did the `sep` change here? In the end, it doesn't matter, since we'll output "protocol-flags%a", and using `Fmt.(list ~sep:nop (append (any " ") (of_to_string Protocol_flag.to_string))` applied to `flags`. So there won't be ever a `~sep` being printed.

Reply:
The `sep` was wrong as it would have output a newline instead of a comma between the configuration directives. Thankfully, it would only ever output one configuration directive. So changing `sep` here does not actually change anything, but it is the morally correct code as we should, in this context, use commas to separate configuration directives.

Reply:
With this in mind maybe we should not have `~sep` be optional...